src/app/clusters/resource-monitoring-server/tests/TestResourceMonitoringCluster.cpp

```diff
@@ -0,0 +1,200 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/resource-monitoring-server/ResourceMonitoringCluster.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h>
+#include <pw_unit_test/framework.h>
+
+#include <app/DefaultSafeAttributePersistenceProvider.h>
+#include <app/clusters/testing/AttributeTesting.h>
+#include <app/clusters/testing/ClusterTester.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <app/server-cluster/testing/TestEventGenerator.h>
+#include <app/server-cluster/testing/TestServerClusterContext.h>
+#include <clusters/ActivatedCarbonFilterMonitoring/Attributes.h>
+#include <clusters/ActivatedCarbonFilterMonitoring/Metadata.h>
+#include <clusters/HepaFilterMonitoring/Attributes.h>
+#include <clusters/HepaFilterMonitoring/Metadata.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ResourceMonitoring;
+using namespace chip::app::Clusters::ResourceMonitoring::Attributes;
+using namespace chip::Test;
+using chip::Test::TestServerClusterContext;
+
+constexpr std::bitset<4> kResourceMonitoringFeatureMap{
+    static_cast<uint32_t>(ResourceMonitoring::Feature::kCondition) | static_cast<uint32_t>(ResourceMonitoring::Feature::kWarning) |
+    static_cast<uint32_t>(ResourceMonitoring::Feature::kReplacementProductList)
+};
+
+namespace {
+
+class ScopedSafeAttributePersistence
+{
+public:
+    ScopedSafeAttributePersistence(TestServerClusterContext & context) : mOldPersistence(app::GetSafeAttributePersistenceProvider())
+    {
+        VerifyOrDie(mPersistence.Init(&context.StorageDelegate()) == CHIP_NO_ERROR);
+        app::SetSafeAttributePersistenceProvider(&mPersistence);
+    }
+    ~ScopedSafeAttributePersistence() { app::SetSafeAttributePersistenceProvider(mOldPersistence); }
+
+private:
+    app::SafeAttributePersistenceProvider * mOldPersistence;
+    app::DefaultSafeAttributePersistenceProvider mPersistence;
+};
+
+class ImmutableReplacementProductListManager : public chip::app::Clusters::ResourceMonitoring::ReplacementProductListManager
+{
+public:
+    CHIP_ERROR Next(ReplacementProductStruct & item) override
+    {
+        if (mIndex >= kReplacementProductListMaxSize)
+        {
+            return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+        }
+
+        switch (mIndex)
+        {
+        case 0: {
+            TEMPORARY_RETURN_IGNORED item.SetProductIdentifierType(ResourceMonitoring::ProductIdentifierTypeEnum::kUpc);
```

Comment:
temporary ignores in new code seem off. In tests lets use `VerifyOrDie(... == CHIP_NO_ERROR)`